tests/system/src/test/java/test/robot/javafx/web/PointerEventTest.java

```diff
@@ -0,0 +1,229 @@
+ * testLeftMiddleRightButtonDrag- Press Left, Middle and Right Buttons and drag
+ */
+
+public class PointerEventTest {
+
+    private static final int LEFT_BUTTON_DRAG = 1;
+    private static final int MIDDLE_BUTTON_DRAG = 4;
+    private static final int RIGHT_BUTTON_DRAG = 2;
+    private static final int SCENE_WIDTH = 250;
+    private static final int SCENE_HEIGHT = SCENE_WIDTH;
+    // Sleep time between mouseMove and element access
+    private static final int SLEEP_TIME = 500;
+
+    private final int DRAG_DISTANCE = 15;
+    private final int DX = 125;
+    private final int DY = 125;
+
+    private static CountDownLatch loadLatch;
+    private static CountDownLatch startupLatch;
+
+    static Document document;
+    static Element element;
+    static Robot robot;
+    static WebView webView;
+    static WebEngine webEngine;
+
+    static volatile Stage stage;
+    static volatile Scene scene;
+
+    private String buttonMask;
+    private int result = 0;
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(Stage primaryStage) {
+            robot = new Robot();
+            stage = primaryStage;
+            stage.setTitle("Mouse Drag Test");
+            webView = new WebView();
+            webEngine = webView.getEngine();
+            String URL =  this.getClass().getResource("pointerEvent.html").toString();
+            webEngine.load( URL);
+            webView.getEngine().getLoadWorker().stateProperty().addListener((ov, o, n) -> {
+                if (n == Worker.State.SUCCEEDED) {
+                    document = webEngine.getDocument();
+                    element = document.getElementById("buttonPressed");
+                    loadLatch.countDown();
+                }
+            });
+            scene = new Scene(new StackPane(webView), SCENE_WIDTH, SCENE_HEIGHT);
+            stage.setScene(scene);
+            stage.setAlwaysOnTop(true);
+            Platform.runLater(startupLatch::countDown);
+            stage.show();
+        }
+    }
+
+    public String mouseButtonDrag(MouseButton... button) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int)(scene.getWindow().getX() + scene.getX() + DX),
+                        (int)(scene.getWindow().getY() + scene.getY() + DY ));
+            robot.mousePress(button);
+        });
+
+        for (int i = 0; i < DRAG_DISTANCE; i++) {
+                final int c = i;
+                Util.runAndWait(() -> {
+                    // Move the mouse backwards so that the pointer does not stay on the popup,if any.
+                    robot.mouseMove((int)(scene.getWindow().getX() + scene.getX() + DX - c),
+                        (int)(scene.getWindow().getY() + scene.getY() + DY) );
+            });
+        }
+
+        Util.sleep(SLEEP_TIME);
+        Util.runAndWait(()  -> {
+            buttonMask = element.getTextContent();
+            robot.mouseRelease(button);
+        });
+
+        return buttonMask;
```

Comment:
if you move the `Integer.parseInt` from the caller to here (and change the return type to `int`) it will simplify the callers a bit. Just a suggestion.

Reply:
changes done as per the comment.